From e7af31ee1cc203eec5e8a6b077604c4f332917cb Mon Sep 17 00:00:00 2001
From: AnNDP <anndp@fsoft.com.vn>
Date: Wed, 6 Jun 2018 02:56:18 -0400
Subject: [PATCH 5/9] waylandsink: Ensure wayland display create queue event
 This patch ensure the complete porting from Gstreamer 1.2.3 commit
 13dcc91b4ee47b3dacc9131738902c022650425d

Signed-off-by: AnNDP <anndp@fsoft.com.vn>
---
 ext/wayland/wldisplay.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ext/wayland/wldisplay.c b/ext/wayland/wldisplay.c
index 39782bc..db1a368 100644
--- a/ext/wayland/wldisplay.c
+++ b/ext/wayland/wldisplay.c
@@ -310,6 +310,9 @@ gst_wl_display_new_existing (struct wl_display * display,
   self->own_display = take_ownership;
 
   self->queue = wl_display_create_queue (self->display);
+  if (!self->queue) {
+      return NULL;
+  }
   self->registry = wl_display_get_registry (self->display);
   wl_proxy_set_queue ((struct wl_proxy *) self->registry, self->queue);
   wl_registry_add_listener (self->registry, &registry_listener, self);
-- 
1.9.1

